dataherald/api/fastapi.py

```diff
@@ -312,3 +315,49 @@ def get_golden_records(self, page: int = 1, limit: int = 10) -> List[GoldenRecor
         start_idx = (page - 1) * limit
         end_idx = start_idx + limit
         return all_records[start_idx:end_idx]
+
+    @override
+    def add_instruction(
+        self, db_connection_id: str, instruction_request: InstructionRequest
+    ) -> Instruction:
+        instruction_repository = InstructionRepository(self.storage)
+        instruction = Instruction(
+            instruction=instruction_request.instruction,
+            db_connection_id=db_connection_id,
+        )
+        return instruction_repository.insert(instruction)
+
+    @override
+    def get_instructions(
+        self, db_connection_id: str, page: int = 1, limit: int = 10
+    ) -> List[Instruction]:
+        instruction_repository = InstructionRepository(self.storage)
+        instructions = instruction_repository.find_by({"db_connection_id": db_connection_id})
+        start_idx = (page - 1) * limit
+        end_idx = start_idx + limit
```

Comment:
These two lines should be part of the `find_by` we should't return a big collection and then just create a sub-list.